search-app/src/cmr/search/services/query_service.clj

```diff
@@ -301,6 +304,102 @@
      (ph/provider-holdings->string
        (:result-format params) provider-holdings {:echo-compatible? (= "true" echo-compatible)})]))
 
+(defn- get-collections-with-deleted-revisions
+  "Executes elasticsearch searches to find collections that have deleted revisions.
+   Returns a list of concept-ids of the collections."
+  [context params]
+  ;;1/ Find all collection revisions that are deleted satisfy the original search params
+  (let [query (make-concepts-query context :collection params)
+        condition (gc/and-conds
+                   [(:condition query)
+                    (qm/boolean-condition :deleted true)])
+        results (common-idx/execute-query context
+                                          (-> query
+                                              (assoc :all-revisions? true)
+                                              (assoc :condition condition)
+                                              (assoc :page-size :unlimited)))
+        coll-concept-ids (map #(first (:concept-id (:fields %)))
+                              (get-in results [:hits :hits]))]
+    (distinct coll-concept-ids)))
+
+(defn- get-visible-collections
+  "Returns the concept ids of collections that are visible from the given collection concept ids"
+  [context coll-concept-ids]
+  (when (seq coll-concept-ids)
+    (let [condition (qm/string-conditions :concept-id coll-concept-ids true)
```

Comment:
We don't have any special code for handling concept-id searches, i.e. this would be the same kind of condition for searching by concept-ids. I tried a test that searches for collections with 2000 concept-ids and it didn't cause any errors. I think we can address this later if we see real problems.